Ocpbugs 55628 - No documentation existing so secondary network bond-cni

kquinn1204 opened the pull request:
<!--- PR title format: [GH#<gh-issue-id>][BZ#<bz-issue-id>][OCPBUGS#<jira-issue-id>][OCPBUGS-55628]: No documentation existing so secondary network bond-cni

<!--- If your changes apply to the latest release and/or in-development version of OpenShift, open your PR against the `main` branch.
Do not create or rename a top-level directory (or any subdirectory in a directory that contains a hugebook.flag file) in the repository and topic map without checking with a docs program manager first.
If a book is being created or modified, there are changes on the Customer Portal that must also be made.

* For more details about the information requested in this template, see:
  https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/create_or_edit_content.adoc#submit-PR --->

Version(s): 4.14+
<!--- Specify the version or versions of OpenShift your PR applies to. -->

Issue:https://issues.redhat.com/browse/OCPBUGS-55628
<!--- Add a link to the Bugzilla, Jira, or GitHub issue, if applicable. --->

Link to docs preview: https://93993--ocpdocs-pr.netlify.app/openshift-enterprise/latest/networking/multiple_networks/secondary_networks/creating-secondary-nwt-other-cni.html
https://93993--ocpdocs-pr.netlify.app/openshift-enterprise/latest/networking/multiple_networks/understanding-multiple-networks.html
<!--- Add direct link(s) to the exact page(s) with updated content from the preview build. --->

QE review:
- [x] QE has approved this change.
<!--- QE approval is required to merge a PR except for changes that do not impact the meaning of the docs. --->

Additional information:
<!--- Optional: Include additional context or expand the description here.--->

<!--- After you open your PR, ask for review from the OpenShift docs team:
  For community authors: Tag @openshift/team-documentation in a GitHub comment.--->

kquinn1204 committed:
OCPBUGS-55628 add bond-cni

mlguerrero12 commented on modules/nw-multus-bond-cni-object.adoc:
```diff
@@ -0,0 +1,96 @@
+|Specifies the arp link monitoring frequency in milliseconds.
+
+|`mtu`
+|`string`
```
I don't see the mode parameter

kquinn1204 commented on modules/nw-multus-bond-cni-object.adoc:
```diff
@@ -0,0 +1,88 @@
+
+|`mode`
+|`string`
+|Specifies the policy that governs how the network traffic is distributed and managed across the multiple network interfaces aggregated into a single logical bonded interface. The bonding modes supported are `balance-rr`, `active-backup` and `balance-xor`. 
```
Supported modes where in a  note here https://docs.okd.io/4.18/networking/hardware_networks/using-pod-level-bonding.html#nw-sriov-cfg-creating-bond-network-attachment-definition_using-pod-level-bonding (this is our docs but OKD link)

kquinn1204 commented on modules/nw-multus-bond-cni-object.adoc:
```diff
@@ -0,0 +1,88 @@
+
+|`failOverMac`
+|`integer`
+|Optional: Specifies the `failOverMac` setting for the bond. Should be set to 1 for `active-backup` bond modes. Default is 0.
```
what are the range of options for this ?

kquinn1204 commented:
Supported modes @mlguerrero12 @cgoncalves only doc'd now wrt to SR-IOV here https://docs.okd.io/4.18/networking/hardware_networks/using-pod-level-bonding.html#nw-sriov-cfg-creating-bond-network-attachment-definition_using-pod-level-bonding (this is our docs but OKD link) keeping this other section more generic with only mention of default mode.

cgoncalves commented:
Only "active-backup" mode is supported. If the default in the bond CNI is "balance-rr" (unsupported), maybe do not mention what the default is and/or say only "active-backup" is supported with attached SR-IOV VFs and point to https://docs.okd.io/4.18/networking/hardware_networks/using-pod-level-bonding.html#nw-sriov-cfg-creating-bond-network-attachment-definition_using-pod-level-bonding

kquinn1204 commented:
So @cgoncalves @mlguerrero12 I updated that and added some relevant context up front then there is a link to the SR-IOV config so hope that is OK. Please add a LGTM if happy.

cgoncalves commented:
/lgtm

kquinn1204 commented:
/label peer-review-needed

lpettyjo reviewed (commented):
LGTM!

jeana-redhat commented on modules/nw-multus-bond-cni-object.adoc:
```diff
@@ -0,0 +1,90 @@
+
+:_mod-docs-content-type: REFERENCE
+[id="nw-multus-bond-cni-object_{context}"]
+= Configuration for a bond-cni secondary network
```
```suggestion
= Configuration for a Bond CNI secondary network
```

jeana-redhat commented on modules/nw-multus-bond-cni-object.adoc:
```diff
@@ -0,0 +1,90 @@
+[id="nw-multus-bond-cni-object_{context}"]
+= Configuration for a bond-cni secondary network
+
+The Bond Container Network Interface (Bond-CNI) enables the aggregation of multiple network interfaces into a single logical "bonded" interface within a container, enhancing network redundancy and fault tolerance. Only SR-IOV Virtual Functions (VFs) are supported for bonding with this plugin.
```
```suggestion
The Bond Container Network Interface (Bond CNI) enables the aggregation of multiple network interfaces into a single logical "bonded" interface within a container, enhancing network redundancy and fault tolerance. Only SR-IOV Virtual Functions (VFs) are supported for bonding with this plugin.
```

jeana-redhat commented on networking/multiple_networks/understanding-multiple-networks.adoc:
```diff
@@ -57,6 +57,9 @@ networks in your cluster:
 
 * *bridge*: xref:../../networking/multiple_networks/secondary_networks/creating-secondary-nwt-other-cni.adoc#nw-multus-bridge-object_configuring-additional-network-cni[Configure a bridge-based secondary network] to allow pods on the same host to communicate with each other and the host.
 
+* *bond-cni*: xref:../../networking/multiple_networks/secondary_networks/creating-secondary-nwt-other-cni.adoc#nw-multus-bond-cni-object_configuring-additional-network-cni[Configure a bond-cni secondary network] providing a method for aggregating
+multiple network interfaces into a single logical _bonded_ interface.
```
- terms
- hard-wrapped
- parallelism

```suggestion
* *bond-cni*: xref:../../networking/multiple_networks/secondary_networks/creating-secondary-nwt-other-cni.adoc#nw-multus-bond-cni-object_configuring-additional-network-cni[Configure a Bond CNI secondary network] to provide a method for aggregating multiple network interfaces into a single logical _bonded_ interface.
```

jeana-redhat commented on modules/nw-multus-bond-cni-object.adoc:
```diff
@@ -0,0 +1,90 @@
+
+The Bond Container Network Interface (Bond-CNI) enables the aggregation of multiple network interfaces into a single logical "bonded" interface within a container, enhancing network redundancy and fault tolerance. Only SR-IOV Virtual Functions (VFs) are supported for bonding with this plugin.
+
+The following object describes the configuration parameters for the bond CNI plugin:
```
```suggestion
The following table describes the configuration parameters for the Bond CNI plugin:
```

jeana-redhat commented on modules/nw-multus-bond-cni-object.adoc:
```diff
@@ -0,0 +1,90 @@
+
+|`name`
+|`string`
+|The value for the `name` parameter you provided previously for the CNO configuration.
```
Also, does CNO need a definition? (or is it a typo of CNI? I am a networking dummy :sweat_smile:)

```suggestion
|The value for the `name` parameter that you provided for the CNO configuration.
```

kquinn1204 replied:
Thanks for picking this up looks like in correct description for this field and this is repeated across all CNIs.

jeana-redhat commented on modules/nw-multus-bond-cni-object.adoc:
```diff
@@ -0,0 +1,90 @@
+
+|`miimon`
+|`string`
+|Specifies the arp link monitoring frequency in milliseconds.
```
Need to define arp link?

jeana-redhat commented on modules/nw-multus-bond-cni-object.adoc:
```diff
@@ -0,0 +1,90 @@
+.Bond CNI plugin JSON configuration object
+[cols=".^2,.^2,.^6",options="header"]
+|====
+|Field|Type|Description
```
Should go over the items in the `Description` column and rephrase them so that they all have a parallel structure.

jeana-redhat commented on modules/nw-multus-bond-cni-object.adoc:
```diff
@@ -0,0 +1,90 @@
+
+|`linksInContainer`
+|`boolean`
+|Optional: Specifies if slave links are in container to start. Default is `false` that is look for interfaces on host before bonding.
```
Any possible way to avoid `slave` as a term here? :grimacing:

jeana-redhat reviewed (commented):
Some inconsistent use of the feature name here. [Seems like](https://github.com/k8snetworkplumbingwg/bond-cni) it is meant to be "Bond CNI" in plain text, and `bond-cni` is the package?

Otherwise, just a couple things to look at here before merge.

/remove-label merge-review-in-progress
/remove-label merge-review-needed

abhatt-rh commented:
/cherrypick enterprise-4.20
/cherrypick enterprise-4.19
/cherrypick enterprise-4.18
/cherrypick enterprise-4.17
/cherrypick enterprise-4.16
/cherrypick enterprise-4.15
/cherrypick enterprise-4.14
/cherrypick enterprise-4.14

abhatt-rh commented:
/cherrypick enterprise-4.17

abhatt-rh commented:
/cherrypick enterprise-4.15
